src/index.ts

```diff
@@ -21,29 +21,97 @@ const ID_MULTIHASH_CODE = identity.code
 export const namespace = '/ipns/'
 export const namespaceLength = namespace.length
 
-export interface IPNSRecord {
+export interface IPNSRecordV1 {
```

Comment:
@achingbrain cosmetic: since 2021 we no longer support V1-alone, this is a hybrid  V1+V2, so maybe it will be less confusing if this type (used by default)  indicates that fact in the name?

```suggestion
export interface IPNSRecordV1V2 {
```

It will look less confusing in code, for example type at the end of the line  below conveys that a hybrid V1+V2 record will be returned, instead  of confusing name that indicates V1-only vs  V2-only:

```diff
- export async function createWithExpiration (peerId: PeerId, value: CID | PeerId | string, seq: number | bigint, expiration: string, options?: CreateV2OrV1Options): Promise<IPNSRecordV1>
+ export async function createWithExpiration (peerId: PeerId, value: CID | PeerId | string, seq: number | bigint, expiration: string, options?: CreateV2OrV1Options): Promise<IPNSRecordV1V2>
```